tools/SendBlobs/FundsDistributor.cs

```diff
@@ -0,0 +1,192 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Text;
+using Nethermind.Cli;
+using Nethermind.Consensus;
+using Nethermind.Core;
+using Nethermind.Core.Extensions;
+using Nethermind.Crypto;
+using Nethermind.Evm;
+using Nethermind.Int256;
+using Nethermind.Logging;
+using Nethermind.Serialization.Rlp;
+using Org.BouncyCastle.Utilities.Encoders;
+
+namespace SendBlobs;
+internal class FundsDistributor
+{
+    /// <summary>
+    /// Distribute the available funds from an address to a number of newly generated private keys.
+    /// </summary>
+    /// <param name="nodeManager"></param>
+    /// <param name="chainId"></param>
+    /// <param name="distributeFrom"></param>
+    /// <param name="keysToMake"></param>
+    /// <param name="keyFilePath">Path to a file </param>
+    /// <returns><see cref="IEnumerable{string}"/> containing all the executed tx hashes.</returns>
+    /// <exception cref="Exception"></exception>
+    public async static Task<IEnumerable<string>> DitributeFunds(INodeManager nodeManager, ulong chainId, Signer distributeFrom, uint keysToMake, string keyFilePath, UInt256 maxFee, UInt256 maxPriorityFee)
+    {
+        string? balanceString = await nodeManager.Post("eth_getBalance", distributeFrom.Address, "latest");
+        if (balanceString is null)
+            throw new Exception($"Unable to get balance for {distributeFrom.Address}");
```

Comment:
You never want to throw base exception types. Please always throw more detailed type.